extensions/lifecycle/tasks/LifecycleTask.js

```diff
@@ -478,6 +488,13 @@ class LifecycleTask extends BackbeatTask {
             }
             return store;
         }, {});
+        // if EODM was not explicitly set and current account is LC account
+        // enable EODM rule by default
+        if (isLCUser && filteredRules.Expiration &&
+        filteredRules.Expiration.ExpiredObjectDeleteMarker !== false) {
```

Comment:
I would have preferred if we didn't have to tweak the applicable rules to support this, but instead do the check whenever we're ready to do the expiration of delete markers. (the reason is that this rule has a specific meaning in AWS API, we're extending that meaning to fit another special case for internal purpose which can bring some confusion - not a huge issue, if that requires some refactoring let's keep this for now as I prefer the amount of changes to be minimal).

Reply:
I'll look into refactoring this. I think it'll be an easy refactor, but will not go into this release, but probably the next.
All relevant data is available in the function that sends the expiration object entries. But will need to edit the logic in the function that handles eodm versions